File: GatePay/frontend/src/app/core/services/beneficiary.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models/api-response.model';

export interface BeneficiaryResponse {
  id: number;
  userId: number;
  beneficiaryUserId?: number;
  accountNumber: string;
  accountHolderName: string;
  nickName?: string;
  bankName: string;
  createdAt: string;
}

export interface CreateBeneficiaryRequest {
  accountNumber: string;
  accountHolderName: string;
  nickName?: string;
  bankName?: string;
}

@Injectable({ providedIn: 'root' })
export class BeneficiaryService {
  private apiUrl = `${environment.apiUrl}/beneficiaries`;

  constructor(private http: HttpClient) {}

  getMyBeneficiaries(): Observable<ApiResponse<BeneficiaryResponse[]>> {
    return this.http.get<ApiResponse<BeneficiaryResponse[]>>(this.apiUrl);
  }

  create(data: CreateBeneficiaryRequest): Observable<ApiResponse<BeneficiaryResponse>> {
    return this.http.post<ApiResponse<BeneficiaryResponse>>(this.apiUrl, data);
  }

  delete(id: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.apiUrl}/${id}`);
  }
}

File: GatePay/backend/src/main/java/com/training/paygate/service/BeneficiaryService.java
package com.training.paygate.service;

import com.training.paygate.dto.request.CreateBeneficiaryRequest;
import com.training.paygate.dto.response.BeneficiaryResponse;

import java.util.List;

public interface BeneficiaryService {
    BeneficiaryResponse create(CreateBeneficiaryRequest request, String currentUsername);
    List<BeneficiaryResponse> getMyBeneficiaries(String currentUsername);
    void delete(Long id, String currentUsername);
    void autoSaveBeneficiary(Long userId, String accountNumber, String accountHolderName, Long targetUserId);
}
